src/libs/utility/text.c

```diff
@@ -147,7 +147,7 @@ char * elektraReplace (char const * const text, char const * const pattern, char
 	ELEKTRA_LOG_DEBUG ("Found “%lu” occurrences of “%s” in “%s”", numberOccurrences, pattern, text);
 
 	ssize_t replacementLength = strlen (replacement);
-	ssize_t textLength = elektraStrLen (text);
+	ssize_t textLength = strlen (text) + 1;
```

Comment:
Honestly, I do not know why Clang Analyzer reports the warnings and why using `strlen` instead of `elektraStrLen` fixes the issue. I could also wrap the first and last call of `strncpy` in  

```c
#ifndef __clang_analyzer__
#endif
```

, to disable the static analysis partially if you prefer that. Adding an additional null check after determining the size of `text`:

```c
if (text == NULL)
{
    return NULL;
}
```

works too, but does needlessly increase the cyclomatic complexity to `11`. Another possible fix would be to just remove `elektraReplace` altogether.